### DISCUSSION TITLE
Configuring Remix to watch for changes in directories outside the app folder

### DISCUSSION THREAD

**Discussion by YuliyaMinsk** [remix-run/remix#9004] [Q&A]
Hello Team!

I'm working on organizing my project's folder structure and I'm considering placing components, utils, and other non-page-related directories outside of the app folder. Is this possible to do while still maintaining all the Remix features such as fast refresh and HMR? If so, could you guide me on how to configure Remix to watch for changes in these folders?

Thank you for your prompt response!

**Comment by jrestall** [comment]
Sounds like you might be after a monorepo setup where you can have separate  packages e.g. ui, utils. This is straightforward if using the Remix Vite plugin.

Here's a great example repo from @ovi-ispas. https://github.com/ovi-ispas/remix-pnpm-monorepo

**Reply by YuliyaMinsk** [reply]
Thanks a lot for your help!
